### PR TITLE
map-value-pairs: add map-value-pairs() parser

### PR DESCRIPTION
**Pull request by bazsi**
This patch adds a map-value-pairs() parser that can be used to map existing
name-value pairs to a different set during processing, in bulk.  Normal
value-pairs expressions can be used, just like with value-pairs based
destinations.

For example:

```
parser p_remap_name_values {
	map-value-pairs(
		key('.apache.*' rekey(add-prefix("foo")))
		pair('.apache.foobar.xxx', "bar")
		pair('123', "567")
		pair("username", '($sha1 $username)')
	);
};
```

Signed-off-by: Balazs Scheidler <balazs.scheidler@balabit.com>

### CODE REVIEW

File: modules/map-value-pairs/map-value-pairs.c [hunk 1]
```diff
@@ -0,0 +1,48 @@
+#include "map-value-pairs.h"
```
Comment (kvch) [inline modules/map-value-pairs/map-value-pairs.c]: Please, add the copyright header to this file, too.

Comment (bazsi) [inline modules/map-value-pairs/map-value-pairs.c]: Did this in followup patches.

**Comment by bazsi** [comment]
I have resolved the review comment, so this should be good to go.

I've restarted the travis job two times already, because of java download errors, let's hope it works this time.

**Comment by bazsi** [comment]
So, the god of travis gave me his blessing :)

@kvch I hope you like my copyright message, so this should be ready to roll.

**Comment by presidento** [comment]
@kira-syslogng retest this please.

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Pull request merged**